packages/amplify-graphql-auth-transformer/src/utils/index.ts

```diff
@@ -111,3 +112,23 @@ export const getConfiguredAuthProviders = (config: AuthTransformerConfig): Confi
   };
   return configuredProviders;
 };
+
+/**
+ * util to get allowed roles for field
+ * if we have a rule like cognito private we can remove all other related roles from the field since it has top level
+ * access by the provider
+ */
+export const getReadRolesForField = (acm: AccessControlMatrix, readRoles: Array<string>, fieldName: string): Array<string> => {
+  const hasCognitoPrivateRole = readRoles.some(r => r === 'userPools:private') && acm.isAllowed('userPools:private', fieldName, 'read');
+  const hasOIDCPrivateRole = readRoles.some(r => r === 'oidc:private') && acm.isAllowed('oidc:private', fieldName, 'read');
+  let allowedRoles = new Array(...readRoles);
```

Comment:
For consistency with the code below, and because the `Array` constructor has an overload that specifies the array size (it makes this line a little less obvious on first glance).

```suggestion
  let allowedRoles = new [...readRoles];
```